torch/_inductor/triton_heuristics.py

```diff
@@ -398,11 +405,22 @@ def run(self, *args, grid, stream):
             launcher.config.pre_hook(
                 {**zip(self.arg_names, args), **launcher.config.kwargs}
             )
-        return launcher(
-            *args,
-            grid=grid,
-            stream=stream,
-        )
+
+        # guard the record_function_ctx and only call it if profiling is currently
+        # in progress, to reduce latency when profiler is not turned on.
+        if autograd_profiler._is_profiler_enabled:
```

Comment:
where is `_is_profiler_enabled` defined, just curious ?

Reply:
I added it in the previous PR in the stack (in torch/autograd/profiler.py)